### ISSUE TITLE
Doxygen replaces # by ::

### ISSUE THREAD

**Issue by Mario-Klebsch** [doxygen/doxygen#10949]
I have this following source code test.cpp:

```
/** a#b */
void foo(){}
```

and the following Doxyfile:

```
GENERATE_XML=YES
GENERATE_HTML=NO
GENERATE_LATEX=NO
```

But the text is changed, "#" gets changed to "::":
```
$ grep para xml/test*cpp.xml
<para>a::b </para>
$
```
I tried doxygen version 1.9.8

I tried to search doxygen documentation for any special meaning of '#'-Chars, but did not find any usefull information. :-(

**Comment by albert-github** [comment]
Also happen is the HTML output.
Tested with 1.12.0 (b3c28ea6cc7ee2f7ece9c3f7539ca01cc8b36e5f)

**Comment by doxygen** [comment]
The special meaning of `#` is listed [here](https://www.doxygen.nl/manual/autolink.html) in particular the note:

```
Note 3:
    For Javadoc compatibility a # may be used instead of a :: in the patterns above. 
```

See also the Javadoc documentation for [`@see`](https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@see) and [`@link`](https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#{@link})

Maybe we could constrain the special meaning of `#` to the commands `{@link ...}` and `@see ...` but this may break existing usage.

**Comment by Mario-Klebsch** [comment]
The Documentation 'Automatic link generation' describes that certain word are automatically converted to a links:

> Links to functions are created if one of the following patterns is encountered:
> 1.   \<functionName>"("\<argument-list>")"
> 2.    \<functionName>"()"
> 3.    "::"\<functionName>
> 4.    (\<className>"::")n\<functionName>"("\<argument-list>")"
> 5.    (\<className>"::")n\<functionName>"("\<argument-list>")"\<modifiers>
> 6.    (\<className>"::")n\<functionName>"()"
> 7.    (\<className>"::")n\<functionName>

and

> Note 3:
>    For Javadoc compatibility a # may be used instead of a :: in the patterns above. 

My understanding of these statements is, that in those pattern lists, '#'can be used instead of '::', so that a#b would match a method a::b() and generate an apropriate link to it,

It does not tell anything about what will happen, when there is no method called a::b(), as in my example.

I would suggest to 
* either limit the effect of '#'->'::' translation to exactly the documented context (with or without replacing the '#' for the generated link text, in case the link target is found

or to 
* document the replacement of '#' and the method of escaping with '%' (including the method of escaping '%') in some general documentation section.

But I am just an occasional doxygen user, so experienced users might think different about it.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10967

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
